frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -705,14 +705,16 @@ bool TypeInference::canCastBetween(const IR::Type* dest, const IR::Type* src) co
             return f->size == 1 && !f->isSigned;
         } else if (auto de = dest->to<IR::Type_SerEnum>()) {
             return TypeMap::equivalent(src, getTypeType(de->type));
+        } else if (auto ii = dest->to<IR::Type_InfInt>()) {
+            return true;
         }
     } else if (src->is<IR::Type_Boolean>()) {
         if (dest->is<IR::Type_Bits>()) {
             auto b = dest->to<IR::Type_Bits>();
             return b->size == 1 && !b->isSigned;
         }
     } else if (src->is<IR::Type_InfInt>()) {
-        return dest->is<IR::Type_Bits>();
+        return dest->is<IR::Type_Bits>() || dest->is<IR::Type_Boolean>();
```

Comment:
Where do you check that the value of `src` is `0` or `1` as mentioned in the P4 Spec PR?

Reply:
The check is in the constant folding code.

Reply:
Sorry, I'm dense and not seeing it. Can you point me to a line number?

Reply:
Nevermind. I found it. 

https://github.com/p4lang/p4c/blob/40c458898de221c9902e1febf85ac5d438f57c47/frontends/common/constantFolding.cpp#L804